src/perf/best-practices.md

```diff
@@ -0,0 +1,375 @@
+  explicitly requested with `Clip.antiAliasWithSaveLayer`)
+  so these operations aren’t as expensive as `Opacity`,
+  but clipping is still costly, so use with caution.
+  By default, clipping is disabled (`Clip.none`),
+  so you must explicitly enable it when needed.
+* To create a rectangle with rounded corners,
+  instead of applying a clipping rectangle,
+  consider using the `borderRadius` property offered
+  by many of the widget classes.
+
+---
+
+## Implement grids and lists thoughtfully 
+
+How your grids and lists are implemented
+_might_ be causing performance problems for your app.
+This section describes an important best
+practice when creating grids and lists,
+and how to determine whether your app uses
+excessive layout passes.
+
+### Be lazy!
+
+When building a large grid or list,
+use the lazy builder methods, with callbacks.
+That ensures that only the visible portion of the
+screen is built at startup time.
+
+For more information and examples, check out:
+
+* [Working with long lists][] in the [Cookbook][]
+* [Creating a `ListView` that loads one page at a time][]
+  a community article by AbdulRahman AlHamali
+* [`Listview.builder`][] API
+
+[Cookbook]: {{site.url}}/cookbook
+[Creating a `ListView` that loads one page at a time]: {{site.medium}}/saugo360/flutter-creating-a-listview-that-loads-one-page-at-a-time-c5c91b6fabd3
+[`Listview.builder`]: {{site.api}}/flutter/widgets/ListView/ListView.builder.html
+[Working with long lists]: {{site.url}}/cookbook/lists/long-lists
+
+<a href="minimize-layout-passes-caused-by-intrinsic-operations"></a>
+### Minimize layout passes caused by intrinsic operations
+
+If you've done much Flutter programming, you are
+probably familiar with [how layout and constraints work][]
+when creating your UI and you might have memorized Flutter's
+basic layout rule: **Constraints go down. Sizes go up.
+Parent sets position.**
+
+For some widgets, particularly grids and lists,
+the layout process can be expensive.
+For example, consider a 10x500 grid of `Card`s.
+A grid should have uniformly sized cells,
+so the layout code performs a pass,
+starting from the root of the grid (in the widget tree),
+asking each card in the visible grid to return
+its _intrinsic_ size&mdash;the size
+that the widget prefers, assuming no constraints.
+With this information,
+the framework calculates a uniform cell size,
+and re-visits the grid objects,
+_telling_ each card what size to use. 
+
+If your code performs unnecessary layout passes,
+it can slow performance. To determine if this
+is your issue, enable the **Track layouts option**
+in DevTools (disabled by default),
+and look at the app's [stack trace][]
+to learn how many layout passes were performed.
+[PENDING: Where is the track layout option in DevTools?]
+
+[PENDING: How does once determine "excessive"?]
+[PENDING: Is the primary way to fix this by lazy loading
+the grid or list? Or is there something else?]
```

Comment:
lazy loading wouldn't help for the intrinsics part. You need to determine if you actually need the intrinsic sizing. For the use case described above (sizing children based on the largest size) you can't get around it. If you can determine the size of your children upfront, though, you can avoid intrinsics by just setting the children to a fixed size.

If you want to layout children relative to the size of another child you can also avoid intrinsics by writing a custom render object that lays out that child anchor first and lays out the other children around it. That can be done in a single pass.

Reply:
done